tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -29,6 +29,20 @@ namespace MongoDB.Driver.Tests
     public class PipelineDefinitionBuilderTests
     {
         // public methods
+        [Fact]
+        public void Create_should_throw_exception_when_output_type_does_not_match_last_stage_type()
```

Comment:
I think it's better to place this test in PipelineStagePipelineDefinitionTests since this test class was designed for PipelineStagePipelineDefinition<TInput, TOutput> :)

Reply:
Actually I think this test should be in `PipelineDefinitionTests.cs` since the code being tested is in the `PipelineDefinition` class.

In fact there is already a test about this in that file. Look for:

    Constructor_should_verify_the_inputs_and_outputs_of_the_stages_and_throw_when_invalid

But the existing test only checks for a mismatch in an intermediate stage (not the final stage) and doesn't assert against the error message.

Look at refactoring the existing test and adding a new one to test the final stage.

Reply:
NOTE: PipelineDefinitionTests.cs file is equal to PipelineStagePipelineDefinitionTests .net class

Reply:
Done.